src/lib/stream.rs

```diff
@@ -278,6 +320,25 @@ impl<'a> CodedInputStream<'a> {
         self.read_raw_varint64().map(|v| v as u32)
     }
 
+    #[cfg(target_endian = "little")]
+    pub fn read_raw_little_endian32(&mut self) -> ProtobufResult<u32> {
+        let mut bytes = [0u8; 4];
+        try!(self.read_raw_bytes_immediate(&mut bytes));
+        Ok(
+            unsafe { mem::transmute(bytes) }
+        )
+    }
+
+    #[cfg(target_endian = "little")]
+    pub fn read_raw_little_endian64(&mut self) -> ProtobufResult<u64> {
+        let mut bytes = [0u8; 8];
+        try!(self.read_raw_bytes_immediate(&mut bytes));
+        Ok(
+            unsafe { mem::transmute(bytes) }
```

Comment:
Is it safe to transmute like that? `bytes` array is guaranteed to be aligned properly?

I'd write it like this:

```
let mut r = 0u64;
let mut bytes = unsafe { mem::transmute(&mut r) };
...
r
```

BTW, it could probably be a little faster if data is initialized with `mem::uninitialized()`.